theano/gpuarray/pool_grad_grad.c

```diff
@@ -16,9 +17,11 @@ KERNEL void pool_grad_grad_kernel(const ga_size nthreads,
     const ga_size ph = (index / pooled_width) % pooled_height;
     const ga_size c = (index / pooled_width / pooled_height) % channels;
     const ga_size n = (index / pooled_width / pooled_height / channels);
-    const ga_size hstart = ph*stride_h;
+    ga_int hstart = static_cast<ga_int>(ph*stride_h) - static_cast<ga_int>(pad_h);
```

Comment:
At the beginning `ph` will be zero and subtracting from unsigned int results in `MAX_INT - 1` value which will fail `max(result, 0)`. That's why I am using casting here.